launch_testing/launch_testing/__init__.py

```diff
@@ -78,26 +92,63 @@ def get_description(self):
 
     def check(self):
         output_lines = self.stdout_data.getvalue().splitlines()
-        assert self.matched, \
-            'Example output (%r) does not match expected output (%r)' % \
-            (output_lines, self.expected_lines)
+        if not self.matched:
+            raise UnmatchedOutputError(
+                'Example output (%r) does not match expected output (%r)' %
+                (output_lines, self.expected_lines))
+
+
+def get_default_filtered_prefixes():
+    return [
+        b'pid', b'rc',
+    ]
+
+
+def get_rmw_output_filter(rmw_implementation):
+    prefix_with_resource = ament_index_python.has_resource(
+        'rmw_output_filter', rmw_implementation)
+    if prefix_with_resource:
+        rmw_filter_output_found = False
+        relative_rmw_filter_output_path = ament_index_python.get_resource(
+            'rmw_output_filter', rmw_implementation)
+        rmw_filter_output_path = os.path.join(
+            prefix_with_resource, 'share', rmw_implementation, relative_rmw_filter_output_path)
+        if os.path.isfile(rmw_filter_output_path):
+            with open(rmw_filter_output_path, 'r') as h:
+                rmw_filter_output = h.read()
+                rmw_filter_output_lines = rmw_filter_output.strip().split('\n')
+                additional_filtered_prefixes = [
+                    str.encode(l) for l in rmw_filter_output_lines]
```

Comment:
Instead of reencoding the previously decoded content open the file in binary mode.